boolean_parser/mixins/sqla.py

```diff
@@ -86,7 +87,12 @@ def _get_field(self, modelclass, field_name, base_name=None):
         field = None
         # Handle hierarchical field names such as 'parent.name'
         if base_name:
-            if base_name in modelclass.__tablename__:
+            # Match alias name
+            if isinstance(modelclass, AliasedClass) and base_name==modelclass._aliased_insp.name:
+                field = getattr(modelclass, field_name, None)
+
+            # Match table name
+            if not field and base_name in modelclass.__tablename__:
                 field = getattr(modelclass, field_name, None)
         else:
```

Comment:
This logic seems reasonable to me.  It seems to me it should also be available when there is no `base_name` supplied, i.e. in the `else` block as well.  I can imagine cases where there is only one model that is aliased, and you only specify a name (of an attribute) on that class, instead of `parent.name`.

Reply:
The aliased class will function just like the modelclass in the else block (no base_name). The only additional behavior that was needed is to compare base_name against the alias name (modelclass._aliased_insp.name).

Reply:
Ok, that makes sense.  I haven't used aliased models that often.